pycernan/avro/client.py

```diff
@@ -1,47 +1,133 @@
 """
     Base Avro client from which all other clients derive.
 """
+import contextlib
 import socket
 
 import pycernan.avro.config
 
 from abc import ABCMeta, abstractmethod
+from queue import Queue, Empty
 
-from pycernan.avro.exceptions import EmptyBatchException
+from pycernan.avro.exceptions import EmptyBatchException, EmptyPoolException
 from pycernan.avro.serde import serialize
 
 
+DefunctConnection = None
```

Comment:
Assuming this should remain an internal implementation detail, suggest naming it with a leading underscore.